File: app/src/main/java/com/dante/diary/net/AppApi.java
package com.dante.diary.net;


import com.dante.diary.model.AppInfo;

import okhttp3.ResponseBody;
import retrofit2.http.GET;
import retrofit2.http.Streaming;
import retrofit2.http.Url;
import rx.Observable;

/**
 * Created by yons on 17/1/3.
 */

public interface AppApi {

    @GET("app.json")
    Observable<AppInfo> getAppInfo();

    @GET
    @Streaming
    Observable<ResponseBody> download(@Url String apkUrl);

}

File: app/src/main/java/com/example/myapplication/net/AppApi.kt
package com.example.myapplication.net

import com.example.myapplication.data.model.AppInfo
import kotlinx.coroutines.flow.Flow
import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Streaming
import retrofit2.http.Url

interface AppApi {

    @GET("app.json")
    fun getAppInfo(): Flow<AppInfo>

    @GET
    @Streaming
    fun download(@Url apkUrl: String): Flow<ResponseBody>

}